PyPy build Windows fails

I could not find any remaining open issue regardings builds for PyPy. However PyPy builds on Windows still fail for me. E.g. here for Windows 64 bit PyPy3.7 (https://github.com/maxbachmann/JaroWinkler/runs/6056061378):
```
-- Found PythonInterp: C:/Users/runneradmin/AppData/Local/Temp/cibw-run-mr6kdv6x/pp37-win_amd64/build/venv/Scripts/python.exe (found version "3.7.13")
-- Could NOT find PythonLibs (missing: PYTHON_LIBRARIES) (found version "3.7.13")
-- Found Python: C:/hostedtoolcache/windows/Python/3.10.4/x64/python3.exe (found version "3.10.4") found components: Interpreter Development Development.Module Development.Embed
...
LINK: command "C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1431~1.311\bin\Hostx86\x64\link.exe /nologo jarowinkler\CMakeFiles\_initialize.dir\_initialize.cxx.obj /out:jarowinkler\_initialize.pypy37-pp73-win_amd64.pyd /implib:jarowinkler\_initialize.lib /pdb:jarowinkler\_initialize.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO /EXPORT:PyInit__initialize kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:jarowinkler\_initialize.pypy37-pp73-win_amd64.pyd.manifest" failed (exit code 1104) with the following output:
LINK : fatal error LNK1104: cannot open file 'python37.lib'
```

--- a/skbuild/cmaker.py
+++ b/skbuild/cmaker.py
@@ -528,8 +528,8 @@ class CMaker:
 
             candidate_suffixes = [""]
             candidate_implementations = ["python"]
-            if hasattr(sys, "pypy_version_info"):
-                candidate_implementations += ["pypy-c", "pypy3-c", "pypy"]
+            if sys.implementation.name == "pypy":
+                candidate_implementations[:0] = ["pypy-c", "pypy3-c", "pypy"]
                 candidate_suffixes.append("-c")
 
             candidate_extensions = [".lib", ".so", ".a"]

--- a/skbuild/cmaker.py
+++ b/skbuild/cmaker.py
@@ -529,7 +529,7 @@ class CMaker:
             candidate_suffixes = [""]
             candidate_implementations = ["python"]
             if hasattr(sys, "pypy_version_info"):
-                candidate_implementations = ["pypy-c", "pypy3-c", "pypy"]
+                candidate_implementations += ["pypy-c", "pypy3-c", "pypy"]
                 candidate_suffixes.append("-c")
 
             candidate_extensions = [".lib", ".so", ".a"]

